define = and <> for entity and embeddable types (#128)
jakartaee/query: spec/src/main/asciidoc/chapters/language/expressions.adoc

[[numeric-promotion]]
==== Numeric types and numeric type promotion

> The type assigned to an operator expression depends on the types of its operand expression, which, only in the case of numeric types, need not be identical.

Reviewer: The wording here is confusing. Possible improvement:
```suggestion
The type assigned to an operator expression depends on the types of its operand expression, which must be identical except in the case of numeric types.
```

Author: Yeah, you're right, that was bad; I rewrote it.
